packages/php/blueprint/src/ZipExportedSchema.php

```
@@ -0,0 +1,145 @@
+			die( 'Error : ' . $archive->errorInfo( true ) );
+		}
+
+		$this->clean();
+
+		return $this->destination;
+	}
+
+	protected function get_default_destination_dir() {
+		return WP_CONTENT_DIR . '/uploads/blueprint';
+	}
+
+	protected function find_steps( $step_name ) {
+		$steps = array_filter(
+			$this->schema['steps'],
+			function ( $step ) use ( $step_name ) {
+				return $step['step'] === $step_name;
+			}
+		);
+		if ( count( $steps ) ) {
+			return $steps;
+		}
+		return null;
+	}
+
+	protected function add_resource( $step, $type ) {
+
+		// check if we have any plugins with resource = self/plugins
+		// if not, we should just skip
+		$steps = $this->find_steps( $step );
+		if ( $steps === null ) {
+			return array();
+		}
+
+		$steps = array_filter(
+			$steps,
+			function ( $plugin ) use ( $type ) {
+				if ( $type === 'plugins' ) {
+					return $plugin['pluginZipFile']['resource'] === 'self/plugins';
+				} else if ( $type === 'themes' ) {
+					return $plugin['themeZipFile']['resource'] === 'self/themes';
+				}
+
+				return false;
+			}
+		);
+
+		if ( count( $steps ) === 0 ) {
+			return array();
+		}
+
+		// create 'plugins' dir
+		$this->maybe_create_dir( $this->working_dir . '/' . $type );
+
+		$files = array();
+
+		foreach ( $steps as $step ) {
+			$resource   = $step[ $type === 'plugins' ? 'pluginZipFile' : 'themeZipFile' ];
+			$destination = $this->working_dir . '/' . $type . '/' . $resource['slug'] . '.zip';
+			$plugin_dir  = WP_CONTENT_DIR . '/' . $type . '/' . $resource['slug'];
+			if ( ! is_dir( $plugin_dir ) ) {
+				// if dir does not exist, see if it's a single file plugin
+				$plugin_dir = $plugin_dir . '.php';
+				if ( ! file_exists( $plugin_dir ) ) {
+					// @todo handle error
+					continue;
+				}
+			}
+			$archive = new \PclZip( $destination );
+			$result  = $archive->create( $plugin_dir, PCLZIP_OPT_REMOVE_PATH, WP_CONTENT_DIR . '/' . $type );
+			if ( $result === 0 ) {
+				die( $archive->errorInfo( true ) );
```

Should we throw an error here instead? Is there a reason to die?

---

Thank you for the catch! Updated in [Do not use die() -- throw an exception instead](https://github.com/woocommerce/woocommerce/pull/49763/commits/2b247dfc44be3efa93952ba61819449487a8a5d6)